pkg/actors/targets/workflow/orchestrator/orchestrator.go

```diff
@@ -103,17 +95,29 @@ func (o *orchestrator) InvokeStream(ctx context.Context, req *internalsv1pb.Inte
 
 // DeactivateActor implements actors.InternalActor
 func (o *orchestrator) Deactivate(ctx context.Context) error {
+	if !o.closed.CompareAndSwap(false, true) {
+		return nil
+	}
+
 	o.wg.Add(1)
-	defer o.wg.Done()
 
 	unlock, err := o.lock.ContextLock(ctx)
 	if err != nil {
-		return err
+		o.wg.Done()
+		return fmt.Errorf("failed to deactivate workflow '%s': %w", o.actorID, err)
 	}
 	defer unlock()
+	o.wg.Done()
+
+	o.table.Delete(o.actorID)
+	o.state = nil
+	o.rstate = nil
+	o.ometa = nil
+	o.ometaBroadcaster.Close()
+	o.lock.Close()
+	o.wg.Wait()
+	orchestratorCache.Put(o)
 
-	o.cleanup()
-	log.Debugf("Workflow actor '%s': deactivated", o.actorID)
```

Comment:
Shall we keep this log? Or did I miss you moving it somewhere else?

Reply:
Just went ahead and removed it- there isn't any useful information a developer/someone debugging can get from this information imo, and is simply taking up cycles.